meerk40t/gui/materialmanager.py

```diff
@@ -726,17 +726,176 @@ def on_delete(self, event):
+                            if numeric_value != 0:
+                                self.op_data.write_persistent(section_name, "power", numeric_value * 10)
+                        elif param=="freq":
+                            # khz
+                            if numeric_value != 0:
+                                self.op_data.write_persistent(section_name, "frequency", numeric_value/1000.0)
+                        elif param=="jumpspeed":
+                            if numeric_value != 0:
+                                self.op_data.write_persistent(section_name, "rapid_enabled", True)
+                                self.op_data.write_persistent(section_name, "rapid_speed", numeric_value)
+                        elif param=="starttc":
+                            if numeric_value != 0:
+                                self.op_data.write_persistent(section_name, "timing_enabled", True)
+                                self.op_data.write_persistent(section_name, "delay_laser_on", numeric_value)
+                        elif param=="laserofftc":
+                            if numeric_value != 0:
+                                self.op_data.write_persistent(section_name, "timing_enabled", True)
+                                self.op_data.write_persistent(section_name, "delay_laser_off", numeric_value)
+                        elif param=="polytc":
+                            if numeric_value != 0:
+                                self.op_data.write_persistent(section_name, "timing_enabled", True)
+                                self.op_data.write_persistent(section_name, "delay_polygon", numeric_value)
+                        elif param=="qpulsewidth":
+                            if numeric_value != 0:
+                                self.op_data.write_persistent(section_name, "pulse_width_enabled", True)
+                                self.op_data.write_persistent(section_name, "pulse_width", numeric_value)
+                        else:
+                            # Unknown / unsupported - a significant amount of the parameters
+                            # would require the addition of different things like hatches,
+                            # wobbles or device specific settings
+                            if numeric_value != 0:
+                                if info_box:
+                                    info_box += "\\n"
+                                info_box += f"{param} = {numeric_value}"
+                # Residual information available?
+                if info_box and info_section_name:
+                    self.op_data.write_persistent(info_section_name, "note", info_box)
+
+        except (OSError, RuntimeError, PermissionError, FileNotFoundError):
+            return False
+        if added:
+            self.op_data.write_configuration()
+        return added
+
     def on_import(self, event):
-        print("Importing")
+       #
+        myfile = ""
+        mydlg = wx.FileDialog(
```

Comment:
This is gui level. It might be nice to import things as a CLI process. And do the manipulations potentially from the commandline for what is essentially parsing files.